jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ContentSourceInputStream.java

```diff
@@ -54,8 +54,12 @@ public int read(byte[] b, int off, int len) throws IOException
         {
             if (chunk != null)
             {
-                if (chunk instanceof Content.Chunk.Error error)
-                    throw IO.rethrow(error.getCause());
+                if (Content.Chunk.isError(chunk))
+                {
+                    Content.Chunk c = chunk;
+                    chunk = Content.Chunk.next(chunk); // TODO should we null here and rely on Source to remember state?
```

Comment:
I think we should follow what you suggest in the TODO and rely on the source to remember the state.